src/main/java/com/uid2/admin/vertx/service/EncryptionKeyService.java

```diff
@@ -139,6 +146,43 @@ private void handleRotateMasterKey(RoutingContext rc) {
         }
     }
 
+    private void handleAddSiteKey(RoutingContext rc) {
+        final Optional<Integer> siteIdOpt = RequestUtil.getSiteId(rc, "site_id");
+        if (!siteIdOpt.isPresent()) {
+            return;
```

Comment:
Why is site id optional and what happens if you don't supply it?

Reply:
If there is no site ID supplied - or it can't be parsed as an int - `RequestUtils.getSiteId` will fail the request with a 400.

It might be good to rename `getSiteId` so that it indicates that it will fail the request.